### DISCUSSION TITLE
Calling edge function from Flutter app returns "Invalid JWT"

### DISCUSSION THREAD

**Discussion by agordeev** [supabase/supabase#19623] [Questions]
I have an Edge Function deployed, which I'm trying to call from my Flutter app:

```dart
    final res = await supabase.functions.invoke(
      'pull-changes',
      body: {
        if (lastSyncTime != null)
          'lastSyncTime': lastSyncTime.toIso8601String(),
      },
    );
```

I'm authenticated, the session is valid. However, the invocation returns Unauthorized with a plain string response body: "Invalid JWT". The function itself doesn't even gets called. Also, headers don't include "Authorization" header. It's unclear from docs if I'm responsible to manually set it through `supabaseClient.functions.setAuth(...)`.  But even if I try to do that, I get the same error.

PS: Inspecting the response further, looks like the error comes from Cloudflare. What should I do?

<img width="745" alt="Screenshot 2023-12-12 at 09 39 28" src="https://github.com/supabase/supabase/assets/2883524/1c7561a8-23b7-4454-b723-6c50088ba51a">

**Comment by agordeev** [comment]
Interesting. Works good immediately after signing in. In an hour or so the problem appears.

**Reply by GaryAustin1** [reply]
Try setting your jwt expire time in the dashboard to 5 minutes if you are testing and see if the condition moves up.  If so you have a refresh issue on the user session in your calling code.

![image](https://github.com/supabase/supabase/assets/54564956/d2200e21-cf91-4a04-a1d0-3a0944fe1c0c)

**Reply by agordeev** [reply]
@GaryAustin1  Yes, I'm getting Invalid JWT after 5 mins now (set to 300). How do I fix that though? I use a standard `supabase.functions.invoke()` function, along with pretty default supabase setup.

**Reply by GaryAustin1** [reply]
Sorry, I don't use Flutter, just was guessing the issue based on your symptom.  
I seem to recall seeing some issues around dev environments and refreshing versus production for Flutter, but that was awhile back.  I would search here and in the Supabase flutter repository issues for similar cases.
For some reason your session is not getting refreshed and I don't know how the Flutter clients deal with that.  In JS there is a timer that runs to refresh the sessions before they expire.
